go/saltpack/verify.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package saltpack
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+)
+
+// NewVerifyStream creates a stream that consumes data from reader
+// r.  It returns the signer's public key and a reader that only
+// contains verified data.  If the signer's key is not in keyring,
+// it will return an error.
+func NewVerifyStream(r io.Reader, keyring SigKeyring) (skey SigningPublicKey, vs io.Reader, err error) {
+	s, err := newVerifyStream(r, MessageTypeAttachedSignature)
+	if err != nil {
+		return nil, nil, err
+	}
+	skey = keyring.LookupSigningPublicKey(s.header.SenderPublic)
```

Comment:
The `SenderPublic` field should be the literal public key bytes that we can use for verification. Any reason we look up the verification key in a keyring, instead of constructing it directly? (@maxtaco I notice we did something similar with decryption. Is the idea to support other key types in the future that use identifiers besides their literal public keys?)